src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -106,12 +100,15 @@ public static bool VerifyWithECDsa(byte[] message, byte[] pubkey, byte[] signatu
         }
 
         // This is for solving the hardfork issue in https://github.com/neo-project/neo/pull/3209
-        [ContractMethod(true, Hardfork.HF_Cockatrice, CpuFee = 1 << 15)]
-        public static bool VerifyWithECDsa(byte[] message, byte[] pubkey, byte[] signature, NamedCurve curve)
+        [ContractMethod(true, Hardfork.HF_Cockatrice, CpuFee = 1 << 15, Name = "verifyWithECDsa")]
+        public static bool VerifyWithECDsaV0(byte[] message, byte[] pubkey, byte[] signature, NamedCurveHash curve)
```

Comment:
But why rename it?

Reply:
Because otherwise we have two methods with the same signature?

Reply:
Exactly.

Reply:
Ok I see you replaced the curve enum .

Reply:
Why not include a if condition in the other method?

if (HFCocakrice){   check enum  }

Reply:
I also reviewed about wondering something in this line of reasoning.
I still did not have time to review carefully.

Reply:
>Why not include a if condition in the other method?

Because other method has different parameter name (namedCurveHash vs namedCurve), and parameter name is included into manifest, and manifest is a part of the node state. We need states to be compatible with 3.7.4 mainnet/testnet. With the current version, the "honest" CryptoLib update occurs at the Cockatrice height, the updated contract state is being put into the Management's  storage. So it's one CryptoLib manifest before Cockatrice and another CryptoLib manifest after Cockatrice. We can't change this behaviour.

So we can't just squash these two methods into one.